phoenix-core/src/main/java/org/apache/phoenix/iterate/TableResultIterator.java

```diff
@@ -186,20 +188,45 @@ public Tuple next() throws SQLException {
                 if (lastTuple != null) {
                     ImmutableBytesWritable ptr = new ImmutableBytesWritable();
                     lastTuple.getKey(ptr);
+                    if (scan.getStartRow() != null
+                            && Bytes.compareTo(scan.getStartRow(), HConstants.EMPTY_START_ROW) != 0) {
+                        if (scan.includeStartRow()) {
+                            if (Bytes.compareTo(scan.getStartRow(), ptr.get()) > 0) {
+                                throw new QueryOutOfScanRangeException("Query returned is out of scan range");
+                            }
+                        } else {
+                            if (Bytes.compareTo(scan.getStartRow(), ptr.get()) >= 0) {
```

Comment:
why is equality check required for start and stop both? are both the end exclusive?